client/client.go

```diff
@@ -46,6 +47,33 @@ func (c *RateLimitedClient) Do(req *http.Request) (*http.Response, error) {
 	return resp, nil
 }
 
+func attentiveBackoff(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+	// Retry for rate limits and server errors.
+	if resp != nil && resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode >= 500 {
+		// Check for a 'Retry-After' header.
+		retryAfter := resp.Header.Get("Retry-After")
+		if retryAfter != "" {
+			if retryAfterSeconds, err := time.ParseDuration(retryAfter + "s"); err == nil {
+				return retryAfterSeconds
+			}
+		}
+
+		// If not check if there is a response in the body
+		var data map[string]interface{}
+		json.NewDecoder(resp.Body).Decode(&data)
+		defer resp.Body.Close()
+		retryAfter, ok := data["Retry-After"].(string)
```

Comment:
Have you ran this against your local instance of incident? We don't provide these fields in the response body like that, you'll want to simulate a rate-limit error to get a response that you can then look at.

You can do this by setting a custom limit with the `go run lib/traffic/trafficcmd/main.go` command if you like.

Reply:
so, a lot of head scratching it appears retryablehttp closes the body before it gets passed on to the backoff function. After talking with Lisa we have done the following:

- Added a naive backoff that uses half the rate limit
- Left in a retry mechanism that tries to read a retry_after header
- Will prioritise a cookie to populate the retry_after header

To read the retry after body would require us to use a different client library which we don't currently fancy. This unblocks netflix and gives a path to a sensible end point.

Thoughts?